packages/framework-provider-aws/src/library/read-models-adapter.ts

```diff
@@ -38,14 +45,27 @@ export async function storeReadModel(
   config: BoosterConfig,
   logger: Logger,
   readModelName: string,
-  readModel: ReadModelInterface
+  readModel: ReadModelInterface,
+  expectedCurrentVersion: number
 ): Promise<void> {
-  await db
-    .put({
-      TableName: config.resourceNames.forReadModel(readModelName),
-      Item: readModel,
-    })
-    .promise()
+  try {
+    await db
+      .put({
+        TableName: config.resourceNames.forReadModel(readModelName),
+        Item: readModel,
+        ConditionExpression: 'attribute_not_exists(boosterMetadata.version) OR boosterMetadata.version = :version',
+        ExpressionAttributeValues: {
+          ':version': expectedCurrentVersion,
+        },
+      })
+      .promise()
+  } catch (e) {
+    // The error will be thrown, but in case of a conditional check, we throw the expected error type by the core
+    if (e.name == 'ConditionalCheckFailedException') {
+      throw new OptimisticConcurrencyUnexpectedVersionError(e.message)
+    }
+    throw e
```

Comment:
The change needed in the provider is:

Throw the error `OptimisticConcurrencyUnexpectedVersionError` if the attribute `boosterMetadata.version` is not equal to `expectedCurrentVersion`